util/changes_test.go

```diff
@@ -0,0 +1,93 @@
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/evergreen-ci/utility"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRecursivelySetUndefinedFields(t *testing.T) {
+	assert := assert.New(t)
+
+	type inner struct {
+		Foo string
+	}
+
+	type deep struct {
+		Inner *inner
+	}
+
+	type shape struct {
+		A string
+		B *int
+		C bool
+		D *deep
+	}
+
+	a := shape{
+		A: "a",
+		B: utility.ToIntPtr(1),
+		C: true,
+		D: &deep{
+			Inner: &inner{
+				Foo: "foo",
+			},
+		},
+	}
+	b := shape{
+		A: "b",
+		D: &deep{
+			Inner: &inner{
+				Foo: "bar",
+			},
+		},
+	}
+
+	aPtr := &a
+	bPtr := &b
+
+	reflectedA := reflect.ValueOf(aPtr).Elem()
```

Comment:
This test would benefit from some comments explaining what case each RecursivelySetUndefinedFields call is testing.